pkg/dataobj/planner/physical/printer_test.go

```diff
@@ -0,0 +1,54 @@
+package physical
+
+import "testing"
+
+func TestPrinter(t *testing.T) {
+	t.Run("simple tree", func(t *testing.T) {
+		p := &Plan{}
+
+		limit := p.AddNode(&Limit{id: "limit"})
+		filter := p.AddNode(&Filter{id: "filter"})
+		merge := p.AddNode(&SortMerge{id: "merge"})
+		scan1 := p.AddNode(&DataObjScan{id: "scan1"})
+		scan2 := p.AddNode(&DataObjScan{id: "scan2"})
+		_ = p.AddEdge(Edge{Parent: limit, Child: filter})
+		_ = p.AddEdge(Edge{Parent: filter, Child: merge})
+		_ = p.AddEdge(Edge{Parent: merge, Child: scan1})
+		_ = p.AddEdge(Edge{Parent: merge, Child: scan2})
+
+		repr := PrintAsTree(p)
+		t.Log("\n" + repr)
```

Comment:
I intentionally did not assert the correctness of the printed plan, because very time you change something in the representation of a node, it will impact the output string.

Therefore only logging to console for human validation.